src/P2P/Seed/Seeder.cpp

```diff
@@ -168,13 +183,20 @@ void Seeder::SeedNewConnection()
             std::make_shared<asio::ip::tcp::socket>(*m_pAsioContext)
         ));
         
-        if(Global::GetConfig().IsPeerBlocked(pSocket->GetIPAddress().GetAddress().to_string())) {
-            LOG_TRACE_F("peer is blocked: {}", pPeer);
+        const std::string& ipAddressStr = pSocket->GetIPAddress().GetAddress().to_string();
```

Comment:
to_string() returns a `std::string` not a `const std::string&`, so you shouldn't store it as a reference. If you follow my suggestion to change the IsPeerXXX functions to take in a `const IPAddress&` instead, then you can just store `pSocket->GetIPAddress()` instead, which does return a const reference.